hudi-client/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriter.java

```diff
@@ -95,25 +102,26 @@
   private static Map<String, HoodieMetadataWriter> instances = new HashMap<>();
 
   public static HoodieMetadataWriter instance(Configuration conf, HoodieWriteConfig writeConfig) {
-    try {
-      return new HoodieMetadataWriter(conf, writeConfig);
-    } catch (IOException e) {
-      throw new HoodieMetadataException("Could not initialize HoodieMetadataWriter", e);
+    String key = writeConfig.getBasePath();
+    if (instances.containsKey(key)) {
```

Comment:
I am still not a fan of this caching. specifically, coz it creates extra complexity like this.